components/tidb_query_expr/src/impl_string.rs

```diff
@@ -219,6 +224,88 @@ pub fn rpad(arg: BytesRef, len: &Int, pad: BytesRef, writer: BytesWriter) -> Res
     }
 }
 
+#[rpn_fn(writer, capture = [ctx])]
+#[inline]
+pub fn add_string_and_duration(
+    ctx: &mut EvalContext,
+    arg0: BytesRef,
+    arg1: &Duration,
+    writer: BytesWriter,
+) -> Result<BytesGuard> {
+    let s = std::str::from_utf8(arg0).unwrap();
+    if is_duration(s) {
```

Comment:
Why not to use `tidb_query_datatype::codec::mysql::duration::parse()`?

Reply:
Okay~

done